terraform/aws/addons/external-dns.tf

```diff
@@ -0,0 +1,75 @@
+#####################
+# Deploy ExternalDNS
+#####################
+# Create Namespace for ExternalDNS
+resource "kubernetes_namespace" "external-dns" {
+  metadata {
+    name = "external-dns"
+  }
+}
+
+resource "helm_release" "external-dns" {
+  name       = "external-dns"
+  repository = "https://charts.bitnami.com/bitnami"
+  chart      = "external-dns"
+  version    = "2.20.10"
```

Comment:
I think, we can move all versions to `.tfvars file` or, better, to `locals` in `init.tf` intermediately after [`terraform {}` section](https://github.com/shalb/cluster.dev/blob/15bb21ef1f5a054f17d74a37c133d5efc6e8c527/docs/terraform-code-structure.md#0-inittf).

Also, we can move `backend` section on the top.


```terrafrom
#
# Init
#

terraform {
  backend "_" {}

  required_version = ">= X.Y.Z" # minimum required Terraform version. Use the oldest version if possible.

  required_providers {
    _ = "~> X.Y" # version of the used provider, pin only MAJOR.MINOR. Use the newest version if possible.
  }

locals {
  external-dns_version = "2.20.10"
  iam_assumable_role_external_dns_version = "~> v2.6.0"
}

}
```

Reply:
Related to `inti.tf` structure #72

Reply:
Terraform don't allow set source and it version in modules. For resources all works fine

```terraform
locals {
  module_version   = "~> v2.6.0"
  resource_version = "2.20.10"
}

resource "helm_release" "external-dns" {
  version    = local.resource_version
...
}

module "iam_assumable_role_external_dns" {
  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"  #! <-- Variables may not be used here.
  version                       = local.module_version #! <-- Variables may not be used here.
...
}
```

Well, TF supports [override files](https://www.terraform.io/docs/configuration/override.html), so for modules vars in source can be emulated in next way:

```terraform
# main.tf
module "iam_assumable_role_external_dns" { # Version in init_override.tf
  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
  create_role                   = true
  role_name                     = "qwerty"
  provider_url                  = "https://"
  role_policy_arns              = ["qwrwqr"]
  oidc_fully_qualified_subjects = ["system:serviceaccount:external-dns:external-dns"]
}

# init_override.tf
module "iam_assumable_role_external_dns" { # <-- linter says "missing 'source'"
  version = "~> v2.6.0"
}
```
or
```terraform
# main.tf
module "iam_assumable_role_external_dns" {
  source                        = "" # See init_override.tf file
  create_role                   = true
  role_name                     = "qwerty"
  provider_url                  = "https://"
  role_policy_arns              = ["qwrwqr"]
  oidc_fully_qualified_subjects = ["system:serviceaccount:external-dns:external-dns"]
}

# init_override.tf
module "iam_assumable_role_external_dns" {
  source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
  version = "~> v2.6.0"
}
```
Does not look very nice, IMHO.

Reply:
I'd suggest to keep it here as is untill TF support module versioning by  variables.
At least we are sticking module version to particular cluster.dev release.

Reply:
Postpone to Build All in Container Task (terraform get)

Reply:
Some other exist variants:
1. [terragrunt `generate`](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/69135cb092b1a0dda5237c0831ce4d3eab15ac07/terragrunt.hcl#L23-L35) + terragrunt variable.
2. [Terrafile](https://github.com/coretech/terrafile)